command/operator_debug.go

```diff
@@ -557,9 +571,17 @@ func (c *OperatorDebugCommand) collectAgentHost(path, id string, client *api.Cli
 		host, err = client.Agent().Host("", id, nil)
 	}
 
-	path = filepath.Join(path, id)
-	c.mkdir(path)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("%s/%s: Failed to retrieve agent host data, err: %v", path, id, err))
 
+		if strings.Contains(err.Error(), structs.ErrPermissionDenied.Error()) {
+			// Drop a hint to help the operator resolve the error
+			c.Ui.Warn(fmt.Sprintf("Agent host retrieval requires agent:read ACL or enable_debug=true.  See https://www.nomadproject.io/api-docs/agent#host for more information."))
```

Comment:
Just a FYI in #9410 we added the ACLs required to the `-help` output as well:

>   If ACLs are enabled, this command will require a token with the 'node:read'
  capability to run. In order to collect information, the token will also
  require the 'agent:read' and 'operator:read' capabilities, as well as the
  'list-jobs' capability for all namespaces.

Reply:
Thanks for the note Tim,  I'll rebase to pull that in.  There is one additional permission requirement though.  To capture pprof's you need agent:write, and in all cases `enable_debug=true` will override the ACL requirement.  Do you think we should mention `enable_debug` in the help text as well?

Reply:
Oh, I missed that. Yeah, let's fix it there.

Reply:
From our discussion this afternoon: let's make a note in the `-help` output about the requirement for `enable_debug`, and then this will be good-to-go.